naas_python/utils/domains_base/authorization.py

```diff
@@ -356,72 +359,65 @@ def _gather_file_credentials(self, credentials_file_path: Path):
         self._file_contents = json.loads(self._file_contents)
         return self._file_contents
 
-    def _generate_credential_file(self, credentials_file_path: Path):
+    def _generate_credential_file(self, credentials_file_path: Path, jupyterhub_token=False, access_token=None, prompt=True):
```

Comment:
I think it would be best to have something like:

```python
def _generate_credential_file(self, credentials_file_path: Path, jupyterhub_access_token=None, prompt=True):

if jupyterhub_access_token is not None:
....
```

Because it seems that anyway in your code, the `access_token` argument will be overwritten if I'm not mistaking.